src/Response/Types/ErrorType.php

```diff
@@ -13,22 +16,209 @@ class ErrorType
     {
         $this->description = $description;
         $this->className = $className;
+        $this->hasErrorTemplate = $hasErrorTemplate;
     }
 
     /**
      * Throws an Api exception from the context provided.
+     *
+     * @param Context $context
      */
     public function throwable(Context $context)
     {
+        $this->updateErrorDescriptionIfHasTemplate($context->getResponse());
+
         return $context->toApiException($this->description, $this->className);
     }
+
+    private function updateErrorDescriptionIfHasTemplate($response): void
+    {
+        $errorDescriptionTemplate = $this->description;
+
+        if ($this->hasErrorTemplate) {
+            $jsonPointersInTemplate = $this->getJsonPointersFromTemplate($errorDescriptionTemplate);
+
+            $errorDescription = $this->updateResponsePlaceholderValues(
+                $errorDescriptionTemplate,
+                $jsonPointersInTemplate,
+                $response
+            );
+
+            $errorDescription = $this->updateHeaderPlaceHolderValues($errorDescription, $response);
+
+            $errorDescription = $this->addPlaceHolderValue(
+                $errorDescription,
+                '{$statusCode}',
+                $response->getStatusCode()
+            );
+
+            $this->description = $errorDescription;
+        }
+    }
+
+    private function updateHeaderPlaceHolderValues(string $errorDescription, ResponseInterface $response): string
+    {
+        $headers = $response->getHeaders();
+        $headerKeys = $this->getHeaderKeys($headers);
+
+        for ($x = 0; $x < count($headerKeys); $x++) {
+            $errorDescription = $this->addPlaceHolderValue(
+                $errorDescription,
+                '{$response.header.' . $headerKeys[$x] . '}',
+                $headers[$headerKeys[$x]],
+                true
+            );
+        }
+
+        return $errorDescription;
+    }
+
+    /**
+     * @param $errorDescription string
+     * @param $jsonPointersInTemplate string[]
+     * @param $response ResponseInterface
+     * @return string Updated error string template.
+     */
+    private function updateResponsePlaceholderValues(
+        string $errorDescription,
+        array $jsonPointersInTemplate,
+        ResponseInterface $response
+    ): string {
+        if (count($jsonPointersInTemplate[0]) < 1) {
+            return $this->addPlaceHolderValue(
+                $errorDescription,
+                '{$response.body}',
+                $response->getRawBody()
+            );
+        }
+
+        $jsonResponsePointer = $this->initializeJsonPointer($response);
+
+        $jsonPointers = $jsonPointersInTemplate[0];
+
+        for ($x = 0; $x < count($jsonPointers); $x++) {
+            $placeHolderValue = "";
+
+            if ($jsonResponsePointer != null) {
+                $placeHolderValue = $this->getJsonPointerValue($jsonResponsePointer, ltrim($jsonPointers[$x], '#'));
```

Comment:
move this null check inside this method and return early if `jsonResponsePointer` is null